LDAR_Sim/src/simulation/simulation_manager.py

```diff
@@ -0,0 +1,298 @@
+        self.in_dir: Path = get_abs_path(self.sim_params[pdc.Sim_Setting_Params.INPUT])
+        self.programs: dict = self.sim_params.pop(pdc.Levels.PROGRAM)
+        self.virtual_world: dict = self.sim_params.pop(pdc.Levels.VIRTUAL)
+        self.output_params: dict = self.sim_params.pop(pdc.Levels.OUTPUTS)
+        self.preseed_random: bool = self.sim_params[pdc.Sim_Setting_Params.PRESEED]
+
+        self._set_methods()
+
+    def _set_methods(self) -> None:
+        self.methods = {
+            method: self.programs[program][pdc.Levels.METHOD][method]
+            for program in self.programs
+            for method in self.programs[program][pdc.Program_Params.METHODS]
+        }
+
+    def setup_properties(self) -> None:
+        self.emis_preseed_val: list[int] = None
+        self.generator_dir = self.in_dir / Generator_Files.GENERATOR_FOLDER
+        self.force_remake_gen: bool = False
+        self.infrastructure: Infrastructure = None
+        self.hash_file_exists: bool = False
+        self.site_measurement_matrix: pd.DataFrame = None
+        self.simulation_count: int = self.sim_params[pdc.Sim_Setting_Params.SIMS]
+        self.sim_start_date: date = date(*self.virtual_world[pdc.Virtual_World_Params.START_DATE])
+        self.sim_end_date: date = date(*self.virtual_world[pdc.Virtual_World_Params.END_DATE])
+        self.seed_timeseries: dict[date, int] = None
+        self.weather: WL = None
+        self.calc_simulation_years()
+
+    def initialize_summary_managers(self) -> None:
+        self.summary_stats_manager: SummaryOutputManager = SummaryOutputManager(
+            output_path=self.out_dir,
+            output_config=self.output_params,
+            sim_years=self.simulation_years,
+            programs=self.programs,
+        )
+
+        self.summary_visualization_manager: SummaryVisualizationManager = (
+            SummaryVisualizationManager(
+                output_config=self.output_params,
+                output_dir=self.out_dir,
+                baseline_program=self.base_program,
+                site_count=self.virtual_world[pdc.Virtual_World_Params.N_SITES],
+            )
+        )
+
+    def check_inputs(self) -> None:
+        check_ERA5_file(self.in_dir, self.virtual_world)
+        has_base: bool = self.base_program in self.programs
+
+        if not (has_base):
+            print(rem.NO_BASE_PROG_ERROR)
+            sys.exit()
+
+    def initialize_outputs(
+        self,
+        input_manager: InputManager,
+        write_parameters: bool = True,
+    ) -> None:
+        if os.path.exists(self.out_dir):
+            shutil.rmtree(self.out_dir)
+        os.makedirs(self.out_dir)
+        if write_parameters:
+            input_manager.write_parameters(self.out_dir / Output_Files.PARAMETER_FILE)
+
+    def check_generator_files(self) -> bool:
+
+        if os.path.exists(self.generator_dir):
+            print(rm.GEN_WARNING_MSG)
+        print(rm.INIT_INFRA)
```

Comment:
Nit: could we move this INIT_INFRA print statement after the pregen/preseed - specifically to be beginning of def setup_virtual_world

Reply:
Will do. This is where it was before though - just a note.